src/rgw/rgw_sal_motr.cc

```diff
@@ -1681,17 +1681,23 @@ MotrObject::~MotrObject() {
 //    return read_op.prepare(dpp);
 //  }
 
-int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string& bname, std::string& key, rgw_obj* target_obj)
+int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string& bname, std::string& key, rgw_obj* target_obj, RGWObjectCtx* rctx)
 {
-  int rc = this->get_bucket_dir_ent(dpp, ent);
+  req_state* s = static_cast<req_state*>(rctx->get_private());
```

Comment:
Check [this](https://github.com/ceph/ceph/pull/44616#issuecomment-1248921018) comment, we can additionally check if req_state is indeed present and then only use it